Library/Homebrew/dependency_collector.rb

```diff
@@ -30,12 +30,18 @@ def initialize
     @requirements = Requirements.new
   end
 
-  def initialize_copy(other)
+  def initialize_dup(other)
     super
     @deps = @deps.dup
     @requirements = @requirements.dup
   end
 
+  def freeze
+    @deps.freeze
```

Comment:
Does `freeze` not freeze instance variables by default? In my head it did but 100% could be wrong.

Reply:
Freezing will stop reassigning instance variables, so you can't do `@deps = ...`.

But it won't stop doing things directly to the object contained in that variable.

For example for arrays, you could still do `@deps << ...` if the array itself isn't also frozen.

```rb
class A
  def m1
    @a = [1]
  end
  
  def m2
    @a << 2
  end
end

a = A.new
p a.m1
p a.m2
p a.freeze
p a.m2 # ok
p a.m1 # error
```

Reply:
Gotcha, makes sense 👍🏻